src/librustc/ty/mod.rs

```diff
@@ -2693,13 +2677,29 @@ fn def_span<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Span {
     tcx.hir.span_if_local(def_id).unwrap()
 }
 
+
+
+/// If the given def ID describes an item belonging to a trait,
+/// return the ID of the trait that the trait item belongs to.
+/// Otherwise, return `None`.
+fn trait_of_item<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Option<DefId> {
```

Comment:
Well there are about 38 more conversions left. How about I just take the first half and you can take the rest? I will start where you left off and take it up until `fn derive_registrar_fn(&self, cnum: CrateNum) -> Option<DefId>;`, and you can start form there. If one of us finishes first and wants to continue we can just touch base on the original issue.